client/image_prune.go

```diff
@@ -22,8 +23,13 @@ func (cli *Client) ImagesPrune(ctx context.Context, pruneFilters filters.Args) (
 		return report, err
 	}
 
-	serverResp, err := cli.post(ctx, "/images/prune", query, nil, nil)
-	if err != nil {
+	cCtx, cancel := context.WithCancel(ctx)
+	RegisterSignalHandler(func() {
+		cancel()
+	})
```

Comment:
I'm not sure I understand how registering the signal handler helps. If the client exits due to `^C`, won't that sever the HTTP connection, and cause the context on the server side to get cancelled anyway?

And `cancel` is an asynchronous operation, so it seems like the client could exit anyway before the HTTP client has a chance to act on the cancellation.